### PR TITLE
Allow using local omp with Apple clang

### PR DESCRIPTION
**Pull request by digantdesai**
Tested as,

```
OMP_NUM_THREADS=8                                          \
TRITON_LOCAL_LIBOMP_PATH="<path..to>/site-packages/torch/" \
CC=$(which clang)                                          \
TRITON_CPU_BACKEND=1                                       \
$(which python3)                                           \
python/tutorials/02-fused-softmax-cpu.py
```

### CODE REVIEW

**Review by minjang — APPROVED** [review]

File: python/triton/runtime/build.py [hunk 1]
```diff
@@ -64,7 +73,20 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
...
+                    cc_cmd += ["-fopenmp"]
+                    cc_cmd += [f"-I{libomp_path}/include"]
+                    cc_cmd += [f"-L{libomp_path}/lib"]
+                    cc_cmd += ["-lomp"]
+                else:
+                    print("Warning: TRITON_LOCAL_LIBOMP_PATH is not set for Apple clang. OpenMP is disabled.")
+            else:
+                cc_cmd += ["-fopenmp"]
+                if libomp_path:
+                    print("Info: Ignoring TRITON_LOCAL_LIBOMP_PATH for non-Apple clang compiler")
```
Comment (minjang) [inline python/triton/runtime/build.py:89]: nit: `print(..., file=sys.stderr)` as a warning?

File: python/triton/runtime/build.py [hunk 2]
```diff
@@ -64,7 +73,20 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
...
+            libomp_path = os.environ.get("TRITON_LOCAL_LIBOMP_PATH", None)
+            if _is_apple_clang():
+                if libomp_path:
+                    cc_cmd += ["-Xclang"]
+                    cc_cmd += ["-fopenmp"]
+                    cc_cmd += [f"-I{libomp_path}/include"]
+                    cc_cmd += [f"-L{libomp_path}/lib"]
+                    cc_cmd += ["-lomp"]
+                else:
+                    print("Warning: TRITON_LOCAL_LIBOMP_PATH is not set for Apple clang. OpenMP is disabled.")
```
Comment (minjang) [inline python/triton/runtime/build.py:85]: What happens in this case? Can triton still compile the kernel and run it, but with a single core?

**Review by minjang — APPROVED** [review]

File: python/triton/runtime/build.py (hunk 2 above)
Comment (digantdesai) [reply python/triton/runtime/build.py:85]: Yes because we don't add `-fopenmp` on cc_cmd

File: python/triton/runtime/build.py (hunk 1 above)
Comment (digantdesai) [reply python/triton/runtime/build.py:89]: Ah good point. I missed this. I will sneak this in with my next PR.

**Comment by int3** [comment]
I know it's meant to be a hack, but a nicer long-term solution would be to handle this at install time

Also I found https://mac.r-project.org/openmp/, we could possibly download it from there

**Comment by digantdesai** [comment]
> [mac.r-project.org/openmp](https://mac.r-project.org/openmp/)
Nice I wish I could have found this. 
Curious, how is it better than using it from torch pip package (which we already depend on)?

**Pull request merged**